jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/javacore/parser/j9/section/threadhistory/ThreadHistoryTagParser.java

```diff
@@ -0,0 +1,75 @@
+/*[INCLUDE-IF JAVA_SPEC_VERSION >= 8]*/
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.dtfj.javacore.parser.j9.section.threadhistory;
+
+import com.ibm.dtfj.javacore.parser.framework.tag.ILineRule;
+import com.ibm.dtfj.javacore.parser.framework.tag.LineRule;
+import com.ibm.dtfj.javacore.parser.framework.tag.TagParser;
+import com.ibm.dtfj.javacore.parser.j9.section.common.CommonPatternMatchers;
+
+/**
+ * Parser rules for lines in the thread history section in the javacore.
+ */
+public class ThreadHistoryTagParser extends TagParser implements
+		IThreadHistoryTypes {
+
+	/**
+	 * Construct a tag parser that handles the tags
+	 * and data from the thread history section of a javacore.
+	 */
+	public ThreadHistoryTagParser() {
+		super(THREADHISTORY_SECTION);
+	}
+
+	/**
+	 * Initialize parser with rules for lines in the thread history section in
+	 * the javacore.
+	 */
+	@Override
+	protected void initTagAttributeRules() {
+		ILineRule lineRule = new LineRule() {
+			@Override
+			public void processLine(String source, int startingOffset) {
+				// The name of the history is on a single line.
+				// 1XECTHTYPE     Current thread history (J9VMThread:0x0000000000725A00)
+				consumeUntilFirstMatch(CommonPatternMatchers.whitespace);
+				addAllCharactersAsTokenAndConsumeFirstMatch(THREADHISTORY_NAME, CommonPatternMatchers.generateMatcher("\\s*\\("));
```

Comment:
Done

Reply:
Using `CommonPatternMatchers.open_paren` instead of `"\\s*\\("` is not equivalent: the former doesn't match whitespace before the opening parenthesis.

Reply:
The consumeUntilFirstMatch will also consume the white space, so CommonPatternMatchers.open_paren is sufficient.
_"The pattern itself is also consumed"_

Reply:
Yes, the use of `consumeUntilFirstMatch()` will consume the spaces before the name, but `addAllCharactersAsTokenAndConsumeFirstMatch()` returns the following characters up to, but not including, the part that matches the given pattern. That pattern doesn't match the space(s) before the opening parenthesis which will then be included in `THREADHISTORY_NAME`.